framework.py

```diff
@@ -9,32 +9,166 @@
 # not expressly granted therein are reserved by Shotgun Software Inc.
 
 import sgtk
+import sys
 import os
+import struct
+from sgtk.util import LocalFileStorageManager
+
 
 class DesktopserverFramework(sgtk.platform.Framework):
+    """
+    Provides browser integration.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(DesktopserverFramework, self).__init__(*args, **kwargs)
+        self._server = None
+        self._settings = None
+        self._tk_framework_desktopserver = None
 
     ##########################################################################################
     # init and destroy
+    def launch_desktop_server(self):
```

Comment:
this forms an explicit contract with desktop. is this an old requirement (looks like something new that was put in)? Design wise, it would be great if desktop didn't know about the framework - could it just start up via `init`? Why do we need this method?